src/main/java/core/basesyntax/strategy/DecrementHandler.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.StorageService;
+
+public class DecrementHandler implements OperationHandler {
+    private final StorageService storageService;
+
+    public DecrementHandler(StorageService storageService) {
+        this.storageService = storageService;
+    }
+
+    @Override
+    public void handle(String fruitName, int quantity) {
+        Fruit fruit = new Fruit(fruitName);
+        storageService.decrement(fruit, quantity);
+    }
```

Comment:
According to the checklist, handling the Purchase operation requires checking the result balance before saving it in the Storage - it should be positive. Throw a `RuntimeException` in case the balance is negative. Please ensure that this logic is implemented to prevent negative balances in the storage.